esi_leap/api/controllers/v1/node.py

```diff
@@ -87,7 +89,20 @@ def get_all(self):
         leases = lease_obj.Lease.get_all({'status': [statuses.CREATED]},
                                          context)
 
+        project_names = {proj.id: proj.name for proj in project_list}
+
+        filtered_nodes = []
         for node in nodes:
+            node_owner = project_names.get(node.owner)
+            node_lessee = project_names.get(node.lessee)
+
+            if resource_class and node.resource_class != resource_class:
+                continue
+            if owner and node_owner != owner:
+                continue
+            if lessee and node_lessee != lessee:
+                continue
+
```

Comment:
You probably don't want to filter like this; instead, you want to filter the list of original nodes. Above we call `ironic.get_node_list`, which calls an ESI function which ultimately calls https://opendev.org/openstack/python-ironicclient/src/branch/master/ironicclient/v1/node.py#L61. The linked method has parameters to filter by resource_class/owner/lessee, so we just want to use that.

Reply:
Okay sure. Will do that.